### PR TITLE
Options: Add 'label' argument to register_setting

### PR DESCRIPTION
**Pull request by Mamaduka**
A sync ticket for https://github.com/WordPress/gutenberg/pull/59243.

Trac ticket: https://core.trac.wordpress.org/ticket/61023

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/rest-api/endpoints/class-wp-rest-settings-controller.php [hunk 1]
```diff
@@ -237,6 +237,7 @@ protected function get_registered_options() {
 
 			$default_schema = array(
 				'type'        => empty( $args['type'] ) ? null : $args['type'],
+				'label'       => empty( $args['label'] ) ? '' : $args['label'],
```
Comment (Mamaduka) [inline src/wp-includes/rest-api/endpoints/class-wp-rest-settings-controller.php]: @TimothyBJacobs, in Gutenberg shim, we used the `title` key for the schema. Would it make sense to have a direct mapping to the argument here, or is the `title` key reserved for similar data in schemas?

Gutenberg PR: https://github.com/WordPress/gutenberg/pull/59243

Comment (TimothyBJacobs) [reply src/wp-includes/rest-api/endpoints/class-wp-rest-settings-controller.php]: Yeah it should be `title` as the prop name. That is how it's defined in the JSON Schema spec.

Comment (Mamaduka) [reply src/wp-includes/rest-api/endpoints/class-wp-rest-settings-controller.php]: Thanks for the clarification!

**Review by TimothyBJacobs — APPROVED** [review]
Looks good. The `wp-api-generated.js` file will need to be rebuilt before committing.

**Comment by Mamaduka** [comment]
Thanks for the review, @TimothyBJacobs 🙇 

> The wp-api-generated.js file will need to be rebuilt before committing.

I always forget about it. I wish there was a pre-commit hook for it. To update it, I need to run a full PHP test suite, correct?

Running the following command did the trick - `npm run test:php -- --group restapi-jsclient`. Though, I'm not sure why it reshuffled the fixture.

**Comment by TimothyBJacobs** [comment]
Yeah the process is a bit frustrating. Was GB active on your checkout?

**Comment by Mamaduka** [comment]
> Was GB active on your checkout?

No. I've got a separate installation just for core patches.

Edit: I updated fixtures again by running the whole test suite, and it worked.

**Comment by ellatrix** [comment]
@Mamaduka Is this good to merge?

**Comment by Mamaduka** [comment]
@ellatrix, yes, it's good to be merged.

**Comment by ellatrix** [comment]
Could you merge it?

**Comment by ellatrix** [comment]
Should https://github.com/WordPress/gutenberg/pull/61351 be included here or in a separate backport?

**Pull request closed without merging**